src/lambda/local/debugConfiguration.ts

```diff
@@ -240,32 +240,34 @@ export function getArchitecture(
     template: CloudFormation.Template | undefined,
     resource: CloudFormation.Resource | undefined,
     invokeTarget: AwsSamDebuggerConfiguration['invokeTarget']
-): Architecture {
+): Architecture | undefined {
     let arch: string | undefined
     let architectureLocation: string
     if (template) {
-        arch = (CloudFormation.getArrayForProperty(resource?.Properties, 'Architectures', template) ?? ['x86_64'])[0]
+        arch = (CloudFormation.getArrayForProperty(resource?.Properties, 'Architectures', template) ?? [])[0]
         architectureLocation = localize('AWS.generic.template', 'template file')
     } else {
         arch = (invokeTarget as CodeTargetProperties)?.architecture
         architectureLocation = localize('AWS.generic.launchConfig', 'launch configuration')
     }
 
-    const isArch = isArchitecture(arch)
+    if (arch) {
+        const isArch = isArchitecture(arch)
 
-    if (!isArch) {
-        getLogger().warn('SAM Invoke: Invalid architecture. Defaulting to x86_64.')
-        vscode.window.showWarningMessage(
-            localize(
-                'AWS.output.sam.invalidArchitecture',
-                'Invalid architecture specified in {0}. Defaulting to x86_64 architecture for invocation.',
-                architectureLocation
+        if (!isArch) {
+            getLogger().warn('SAM Invoke: Invalid architecture. Defaulting to x86_64.')
+            vscode.window.showWarningMessage(
+                localize(
+                    'AWS.output.sam.invalidArchitecture',
+                    'Invalid architecture specified in {0}. Defaulting to x86_64 architecture for invocation.',
+                    architectureLocation
+                )
             )
-        )
-        arch = 'x86_64'
-    }
+            arch = 'x86_64'
```

Comment:
I'm conflicted over forcing a default. Do we _need_ to set an architecture or will every other part of our code base work just fine if we don't specify one?

Reply:
We can leave it as undefined but obviously we want to have something that's not an invalid runtime. Open to either; SAM will handle undefined as `x86_64`.